File: node/movies_create_table.js
var AWS = require('aws-sdk');

AWS.config.update({
  region: "us-east-1",
  endpoint: "http://127.0.0.1:8000"
  //endpoint: "https://dynamodb.us-east-1.amazonaws.com"
});

var dynamodb = new AWS.DynamoDB();

var params = {
  TableName: "Movies",
  KeySchema: [
    { AttributeName: "year",  KeyType: "HASH" },  // partition key
    { AttributeName: "title", KeyType: "RANGE" }, // sort key
  ],
  AttributeDefinitions: [
    { AttributeName: "year",  AttributeType: "N" }, // number
    { AttributeName: "title", AttributeType: "S" }, // string
  ],
  ProvisionedThroughput: {
    ReadCapacityUnits: 10,
    WriteCapacityUnits: 100
  }
};

dynamodb.createTable(params, function(err, data) {
  if (err) {
    console.error("Could not create table: err =", JSON.stringify(err, null, 2));
  } else {
    console.log("Created table. Table description =", JSON.stringify(data, null, 2));
  }
});

File: go/movies_create_table.go
package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
)

func main() {
	// create an aws session
	sess := session.Must(session.NewSession(&aws.Config{
		Region:   aws.String("us-east-1"),
		Endpoint: aws.String("http://127.0.0.1:8000"),
		//EndPoint: aws.String("https://dynamodb.us-east-1.amazonaws.com"),
	}))

	// create a dynamodb instance
	db := dynamodb.New(sess)

	// create the api params
	params := &dynamodb.CreateTableInput{
		TableName: aws.String("Movies"),
		KeySchema: []*dynamodb.KeySchemaElement{
			{AttributeName: aws.String("year"), KeyType: aws.String("HASH")},
			{AttributeName: aws.String("title"), KeyType: aws.String("RANGE")},
		},
		AttributeDefinitions: []*dynamodb.AttributeDefinition{
			{AttributeName: aws.String("year"), AttributeType: aws.String("N")},
			{AttributeName: aws.String("title"), AttributeType: aws.String("S")},
		},
		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
			ReadCapacityUnits:  aws.Int64(10),
			WriteCapacityUnits: aws.Int64(100),
		},
	}

	// create the table
	resp, err := db.CreateTable(params)
	if err != nil {
		fmt.Println(err.Error())
		return
	}

	// print the response data
	fmt.Println(resp)
}
